TMEDIA 793: Update right rail layouts

JackHowa opened the pull request:
## Description

Update right rail and advanced right rail layout using the new architecture 

## Jira Ticket

- [TMEDIA-793](https://arcpublishing.atlassian.net/browse/TMEDIA-793)

## Acceptance Criteria

- Update the right-rail and right-rail-advanced layouts so they can be used with blocks on the new architecture 

## Test Steps

1. Checkout this branch `git checkout TMEDIA-793-right-rail-layouts`
2. `npm run storybook`
3. Go to right rail page 
4. Compare with mocks figma https://www.figma.com/file/wQd5edf6Nt1MB3b6Qlzp6x/Arc-Blocks-Design-System?node-id=1007%3A1639
5. Check on mobile and desktop using the news theme
6. Go to advanced right rail 
7. Compare the order with the published version https://5f91e4721ccaba0022a10782-oztmgknobz.chromatic.com/?path=/story/layouts-right-rail-advanced--basic
8. Spacing should match our existing standards in right rail. The mocks spacing in figma trumps our previously-published versions. 

## Effect Of Changes

### Before

- More nesting. Unused classes

### After

- Only styles that are used at a breakpoint are included 

## Dependencies or Side Effects

Feature pack pr: https://github.com/WPMedia/arc-themes-feature-pack/pull/235

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.


----
# **Questions:** 

- I'm using the stack in a couple places to allow for customization and gap. However, I did not use the Stack in the instance where I needed to change flex direction. Should that be a behavior of the Stack? I was not able to override the styles via the theme file; based on a breakpoint, we cannot change the props going in to the component.  -> I was using the wrong theme component; I should have been using grid
- Is this method of implementing the layout what was expected for the architecture? Is it ok not to use a theme component?  -> we want to match the mocks rather than the actual 
- Do you have any suggestions for the naming of the interior nested divs of the right rail and the interior main item? -> updated names of fixed width and others where it was matching a pagebuilder name
- I included some `initial` in the values as the breakpoints were not allowing overriding the values. I seem to remember this being fixed. I updated the arc themes components package. I was not seeing the breakpoints updating as expected. Is there an issue with my setup? -> this is only fixed for components, not blocks

## **Notes:** 


- There was some odd behavior with negative margins on the actual layout. I'm still investigating if the actual mocks have the border flush with the bottom of the main item or 24px below it (like the v1 layout) 
<img width="228" alt="Screen Shot 2022-05-09 at 17 08 49" src="https://user-images.githubusercontent.com/5950956/167506786-5848a997-9e3a-4f7e-9f91-0547ad7b1f90.png"> -> border should be flush to the end of the content rather than with the additional 24px
- I have not started work on the advanced layout
- I do not have a feature pack theme token update yet
- I have not updated the tests for this pr -> tests are updated

JackHowa committed:
refactor: Remove unused styles after using the reset in arc themes components

JackHowa committed:
feat: Add arc themes components import and modify peer deps expectation

JackHowa committed:
ci: Include top-level index scss file

JackHowa committed:
feat: Recreating layout of right rail removing news theme css and setting up grid based layout

JackHowa committed:
feat: Add new tokens and using mixins to define the layout of the right rail block

JackHowa committed:
chore: Remove index.js and update entry

JackHowa committed:
fix: Use bigger gap based on previous implementation

JackHowa committed:
chore: Update arc themes components

JackHowa committed:
feat: Add interior logic for flex

JackHowa committed:
feat: Update the items to display dynamically in desktop and mobile

JackHowa committed:
docs: Remove comment from theme file

JackHowa committed:
feat: Add padding so that the bottom border is not flush to the end

JackHowa committed:
docs: Only include globals used for news theme

JackHowa committed:
style: Update order of include

JackHowa committed:
feat: Use grid rather than flex for handling the grid container and add the padding/margin to match the mocks rather than the actual

JackHowa committed:
feat: Add bottom and top margin to the footer if it exists to match the mocks

JackHowa committed:
refactor: Rename header to be navigation as per the pagebuilder name

Per feedback on naming

JackHowa committed:
refactor: Rename main item to be full width 1 per pagebuilder naming

JackHowa committed:
refactor: Rename to use full width 2 rather than extra main item per pagebuilder names

JackHowa committed:
fix: Handle if undefined children. Fix typo.

JackHowa committed:
test: Update snapshots based on new classes. Check for case if there's no navigation

JackHowa commented on blocks/right-rail-block/layouts/right-rail/default.jsx:
```diff
@@ -5,7 +5,7 @@ import { Grid, Stack } from "@wpmedia/arc-themes-components";
 const LAYOUT_CLASS_NAME = "b-right-rail";
 
-const useFeatueList = () => {
+const useFeatureList = () => {
```
just fixed typo

JackHowa committed:
feat: Update right rail advanced to reorder elements. Use display contents to allow nested divs to be reordered.

JackHowa committed:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-793-right-rail-layouts

matthewroach commented on .storybook/themes/news.scss:
```diff
@@ -1 +1,192 @@
+	$alias: (
+		// Layout
+		"layout-max-width": 1440,
+		"content-max-width": 1216,
```
The max-width for News theme is 1440px - this setting is showing a lot of visual regression changes via chromatic

JackHowa replied:
Thanks for looking at this and storybook. Not sure I totally follow your comment. I have the alias set as 1440. What are you suggesting that I do?

JackHowa replied:
I can replace what I have using content-scale-width with 1440 explicitly. is that the goal? or are you talking about nothing growing past 1440

JackHowa replied:
Updated @matthewroach

matthewroach replied:
@JackHowa - I created a separate pull request to show what I was meaning. https://github.com/WPMedia/arc-themes-blocks/pull/1342

The "container" for news themes is 1440px - and then there is spacing either side of the content which means the overall layout width needs to wider for the `content-scale-width` logic to work to apply the responsive space either side of the container

You can see the issue in the Chromatic snapshot from this branch - https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=627ae0fd101555004a8c62bd

and on the new pull request for the same chromatic snapshot you can see the spacing either side of the layout is maintained - https://www.chromatic.com/test?appId=5f91e4721ccaba0022a10782&id=627b79425d891d004aa05e06

JackHowa replied:
ah I see at the 1200 px breakpoint. it looks like the spacing on the sides are improved in the new snapshot
![Screen Shot 2022-05-11 at 07 00 42](https://user-images.githubusercontent.com/5950956/167844918-e144b55e-9c9d-4645-8026-7baa0160e123.png)

JackHowa replied:
merged

JackHowa committed:
fix: Update to max at 1440 px per feedback

https://github.com/WPMedia/arc-themes-blocks/pull/1335/files#r869690533

JackHowa committed:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-793-right-rail-layouts

matthewroach committed:
Update layout widths for news styles

Commit:
Merge pull request #1342 from WPMedia/TMEDIA-793-width-adjustments

TMEDIA-793 - Update layout widths for news styles

ecarlisle commented on .storybook/themes/news.scss:
```diff
@@ -1 +1,188 @@
+					margin: var(--global-spacing-6) 0 0 0,
+				),
+				right-rail-rail-container: (
+					grid-template-columns: initial,
```
Are we still needing to use `initial` for prop values on the default viewport?
The question was asked recently and I want to make sure I have the latest info.
Thanks!

JackHowa replied:
yep this is needed for blocks. I put a note in the pr as well: 

> I included some initial in the values as the breakpoints were not allowing overriding the values. I seem to remember this being fixed. I updated the arc themes components package. I was not seeing the breakpoints updating as expected. Is there an issue with my setup? -> this is only fixed for components, not blocks

ecarlisle replied:
Cool! Sorry I missed the note!

ecarlisle commented on .storybook/themes/news.scss:
```diff
@@ -1 +1,188 @@
+					border-right: initial,
+				),
+				right-rail-footer: (
+					margin: var(--global-spacing-10) 0 var(--global-spacing-8) 0,
```
With components we've been avoiding the use of CSS margins so they can be framed accordingly with a `<Stack>` component. It seems perfect to use margin for layout, especially when margins differ on sides of an element. I wanted to ask to be certain. Thanks!

JackHowa replied:
spacing should be left to layouts -- yep!

ecarlisle commented on blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx:
```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
-				</div>
+		<div className={LAYOUT_CLASS_NAME}>
+			{navigation ? (
+				<Stack role="banner" className={`${LAYOUT_CLASS_NAME}__navigation`}>
```
It's too bad we're not still using a semantic `<header>` here, but you've got ARIA covered by having `role="banner"`. 👍🏼 
Wondering if `<Stack>` could/should have a `tagName` attribute for cases like these, rather than getting the default `<div>` container. 🤔 
https://github.com/WPMedia/arc-themes-components/blob/arc-themes-release-version-2.00/src/components/stack/index.jsx#L18
(thoughts of a different scope)

JackHowa replied:
That's a good question. I think for the scope of this requirement it is not needed.

matthewroach replied:
Why wouldn't we update the `Stack` component to do this? It's a nice addition and keeps our code cleaner

matthewroach replied:
See PR in Components for updated `Stack` - https://github.com/WPMedia/arc-themes-components/pull/94

JackHowa replied:
I'm not sure I understand why this is necessary. Is this an accessibility suggestion? or is this a nice-to-have for the future?  @matthewroach

matthewroach replied:
We should use semantic elements over additional attributes to achieve the same outcome. The first rule of ARIA is not to use ARIA. Also we should be setting a standard here, and I worry having the `role` here would also lead to confusion in the future

ecarlisle commented on blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx:
```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
+						</Stack>
+					</Stack>
+				</Grid>
+				{featureList["7"] > 0 ? (
```
This line with the ternary is the only one preventing full test coverage.
Do we need a test for this?
```
> jest --changedSince=origin/arc-themes-release-version-2.00 --coverage --passWithNoTests

 PASS  blocks/right-rail-block/layouts/right-rail/default.test.jsx (9.772 s)
 PASS  blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.test.jsx (9.775 s)
-------------------------------------------------------|---------|----------|---------|---------|-------------------
File                                                   | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-------------------------------------------------------|---------|----------|---------|---------|-------------------
All files                                              |     100 |    95.83 |     100 |     100 |
 right-rail-advanced-block/layouts/right-rail-advanced |     100 |    91.66 |     100 |     100 |
  default.jsx                                          |     100 |    91.66 |     100 |     100 | 61
 right-rail-block/layouts/right-rail                   |     100 |      100 |     100 |     100 |
  default.jsx                                          |     100 |      100 |     100 |     100 |
-------------------------------------------------------|---------|----------|---------|---------|-------------------
```

JackHowa replied:
updated [6857e5f](https://github.com/WPMedia/arc-themes-blocks/pull/1335/commits/6857e5f7cd3e167a4cd55fbb6e3d250b008729e3) good q!

nschubach replied:
I know this isn't you, but what even is featureList["7"] ?

nschubach replied:
nm, I figured it out... I do not like this code :p

ecarlisle replied:
Cool, got the update. 100% coverage!

nschubach replied:
@JackHowa it's not you, it's the system

ecarlisle reviewed (changes requested):
Layouts look great on mobile and desktop.
Great use of the new components & tokens!
Looks like 1 line is preventing full test coverage.

JackHowa committed:
test: Improve test coverage not rendering full width

ecarlisle commented on .storybook/themes/news.scss:
```diff
@@ -1 +1,188 @@
+					padding: 0 0 0 var(--global-spacing-6),
+				),
+				right-rail-main-interior-item: (
+					border-right: 1px solid #dadada,
```
This hex value isn't one of our neutral color tokens, so assuming that using the hex is fine.

JackHowa replied:
yeah the color wasn't available to one of the neutrals. I can add it to the design inconsistencies list. I don't think this should be a blocker for achieving parity.

```
		"neutral-1": #e6efff,
		"neutral-2": #c3cfe3,
		"neutral-3": #a3b0c6,
		"neutral-4": #8593aa,
		"neutral-5": #69778e,
		"neutral-6": #4f5c71,
		"neutral-7": #384355,
		"neutral-8": #232b39,
		"neutral-9": #10151c,
		"neutral-10": #000000,

nschubach commented on blocks/right-rail-block/package.json:
```diff
@@ -25,9 +24,7 @@
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
```suggestion
}

```

nschubach commented on blocks/right-rail-advanced-block/package.json:
```diff
@@ -25,9 +24,7 @@
   },
   "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
-}
+}
```
```suggestion
}

```

Commit:
Update blocks/right-rail-advanced-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

Commit:
Update blocks/right-rail-block/package.json

Co-authored-by: nschubach <nschubach@gmail.com>

Commit:
Merge branch 'arc-themes-release-version-2.00' into TMEDIA-793-right-rail-layouts

nschubach commented on .storybook/themes/news.scss:
```diff
@@ -1 +1,188 @@
+				),
+				right-rail-rail-container: (
+					grid-template-columns: 2fr 1fr,
+					gap: initial,
```
did you want `initial` or `0`?  `initial` is `normal` and normal is browser specific and has special rules: https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap#syntax

JackHowa replied:
will update

JackHowa replied:
updated -- thanks @nschubach

nschubach reviewed (changes requested):
I only see one thing that I'm concerned about.  Otherwise I don't see anything blocking here.

JackHowa committed:
fix: Update based on browser fallback and rules for gap

Per feedback https://github.com/WPMedia/arc-themes-blocks/pull/1335#pullrequestreview-969971133

matthewroach commented on blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx:
```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item-1`}>{main}</Stack>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item-2`}>{main2}</Stack>
+					</Stack>
+					<Stack role="complementary" className={`${LAYOUT_CLASS_NAME}__main-side-rail`}>
```
`role="complementary` is not needed here

JackHowa replied:
ok will remove. saw "aside" element was used before

JackHowa replied:
removed

matthewroach commented on blocks/right-rail-block/layouts/right-rail/default.jsx:
```diff
@@ -15,36 +17,36 @@ const useFeatueList = () => {
-					</div>
+				<Grid className={`${LAYOUT_CLASS_NAME}__rail-container`}>
+					<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item`}>{main}</Stack>
+					<Stack role="complementary" className={`${LAYOUT_CLASS_NAME}__main-side-rail`}>
```
```suggestion
					<Stack role="complementary" className={`${LAYOUT_CLASS_NAME}__main-right-rail`}>
```

nschubach replied:
accessibility concern?

JackHowa replied:
updated

matthewroach commented on blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx:
```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-middle`}>
+							{rightRailMiddle}
+						</Stack>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-bottom`}>
```
```suggestion
						<Stack className={`${LAYOUT_CLASS_NAME}__main-right-rail-bottom`}>
```

JackHowa replied:
updated

matthewroach commented on blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx:
```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-interior-item-2`}>{main2}</Stack>
+					</Stack>
+					<Stack role="complementary" className={`${LAYOUT_CLASS_NAME}__main-side-rail`}>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-top`}>{rightRailTop}</Stack>
```
```suggestion
						<Stack className={`${LAYOUT_CLASS_NAME}__main-right-rail-top`}>{rightRailTop}</Stack>
```

JackHowa replied:
updated

matthewroach commented on blocks/right-rail-advanced-block/layouts/right-rail-advanced/default.jsx:
```diff
@@ -25,44 +27,46 @@ const RightRailAdvancedLayout = ({ children }) => {
+					</Stack>
+					<Stack role="complementary" className={`${LAYOUT_CLASS_NAME}__main-side-rail`}>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-top`}>{rightRailTop}</Stack>
+						<Stack className={`${LAYOUT_CLASS_NAME}__main-side-rail-middle`}>
```
```suggestion
						<Stack className={`${LAYOUT_CLASS_NAME}__main-right-rail-middle`}>
```

JackHowa replied:
updated

JackHowa committed:
fix: Update to be more explicit of a gap of 0 if rows or columns are added to this forked block

Per Nick discussion: only a concern if someone styles it with { columns: 2; } or something, but it could be an interesting "bug" if columns are used

JackHowa committed:
refactor: Remove role complementary based on feedback

https://github.com/WPMedia/arc-themes-blocks/pull/1335#discussion_r870742889

JackHowa committed:
refactor: Specify direction of class name in the name

Based on feedback https://github.com/WPMedia/arc-themes-blocks/pull/1335#discussion_r870744296

nschubach reviewed (approved):
I don't see anything.  Deferring to Matt's suggestions.

matthewroach committed:
Update Right Rail blocks to use updated Stack as prop
